.github/workflows/test.yml

```diff
@@ -0,0 +1,44 @@
+name: End-to-end tests
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+    defaults:
+      run:
+        working-directory: ./utility-emissions-channel
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '12.15.0'
+      - name: Download Fabric binaries
+        run: |
+          sudo chmod +x /usr/local/bin/docker-compose
+          cd docker-compose-setup/
+          ./bootstrap.sh  2.2.1 1.4.9 -d -s
+      - name: Start local Hardhat network
+        run: |
+          cd ../net-emissions-token-network/
+          npm i -g npm@7.24.1
+          npm ci
+          echo "n" | npx hardhat node --show-accounts &
+      - name: Build utility-emissions-channel project
+        run: |
+          cd docker-compose-setup/
+          echo "127.0.0.1       auditor1.carbonAccounting.com
+          127.0.0.1       auditor2.carbonAccounting.com
+          127.0.0.1       peer1.auditor1.carbonAccounting.com
+          127.0.0.1       peer1.auditor2.carbonAccounting.com
+          127.0.0.1       peer1.auditor1.carbonAccounting.com" | sudo tee -a /etc/hosts
+          ./start.sh &
+      - name: Run tests
+        run: |
+          sleep 240
+          cd typescript_app/
+          npm run test:setup
+          npm run test
```

Comment:
@hs2361 , Nice, CI is passing. have some suggestions

 Can you move these to a separate bash script (something like `ci-typescript_app-test.sh`). Also please add a linter job.

Reply:
Hi, thanks for the review. I feel it would be better to leave the commands within the YAML file itself, since they would then show up as "steps" whenever the CI runs. This might make it simpler for someone to debug a CI failure as they would be able to identify the failing step quicker. Do let me know about your rationale on separating out the commands into a Bash script as well.

Reply:
@Zzocker I think we would need to move to ESLint from TSLint first (#268) before we add in the lint job. For now, I think I can just add in the test setup command to the README file. I'll work on #268 and then open a new issue for adding in the lint job. Does that sound good?

Reply:
I'm okay with having the tests steps execute directly in this step rather than expecting a `ci-typescript_ap-test.sh` script.
If it is required to always run `npm run test:setup` before executing the tests I'd prefer to merge this command in the test script itself. If it is not required to run the setup each time, please add the corresponding information to the readme.

I'd like to merge this PR asap and then extend the pipeline step by step. We should start running all the tests that we already have and then extend the testing with linter, unit tests, etc

Reply:
Hi, @udosson. I think we need to run the setup script only once. I'll add it to the README file.